Automattic/wp-calypso
# Fix the renew link in the expiring domain soon notification

We had some additional code in the shopping cart that adds more strict checks for subscriptions renewals (see D132429-code). Which means that we can't renew a domain using the `domain_map` product slug since it doesn't matches the subscription product. Apparently we already have the `product_slug` returned from the REST API so we only needed to add that in the renew link.

## Proposed Changes

* add the productSlug in the renew link for the domain expiration notice

## Why are these changes being made?

* Currently clicking on the "Renew" link in the expiring soon notice gives a weird error "Sorry, this user account does not own the product you are trying to renew."

## Testing Instructions

* Run this branch and click on the "Renew" link in an expiring soon notification

<img width="663" alt="Screenshot 2024-06-19 at 16 49 35" src="https://github.com/Automattic/wp-calypso/assets/1355045/34e9b1f5-d523-4011-8d07-67da41c2c37a">

* Without this PR you'll see an empty cart and an error

<img width="1510" alt="Screenshot 2024-06-19 at 16 43 22" src="https://github.com/Automattic/wp-calypso/assets/1355045/5e1d9dbf-3d18-420e-bda4-7e530516c457">

* With diff you'll see the domain added in the shopping cart

## Pre-merge Checklist
